### PR TITLE
[ODS-6527] Optimize cached method invocation signature computations to eliminate use of SOH (small object heap)

### PR DESCRIPTION
**Pull request by gmcelhanon**

### CODE REVIEW

File: Application/EdFi.Ods.Tests/EdFi.Ods.Common/Caching/XxHash3CodeTests.cs [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+using EdFi.Ods.Common.Caching;
+
+namespace EdFi.Ods.Tests.EdFi.Ods.Common.Caching;
+
+using NUnit.Framework;
+using Shouldly;
+using Standart.Hash.xxHash;
+
+[TestFixture]
+public class XxHash3CodeTests
```
Comment (axelmarquezh) [inline Application/EdFi.Ods.Tests/EdFi.Ods.Common/Caching/XxHash3CodeTests.cs:17]: Optional: add test coverage for `null`, `ulong`, `long`, `byte[]`, `short`, and `Resource` types.

**Review by axelmarquezh — APPROVED** [review]
Substantial improvements both in execution time and allocations:

image

**Review by axelmarquezh — APPROVED** [review]
